iree/compiler/Conversion/CodegenUtils/GetNumWorkgroups.cpp

```diff
@@ -101,7 +102,9 @@ static Value buildCeilDivConstDenominator(PatternRewriter &rewriter,
 
 LogicalResult createNumWorkgroupsFromResultShape(
     PatternRewriter &rewriter, linalg::LinalgOp linalgOp, FuncOp entryPointFn,
-    llvm::StringRef numWorkgroupsFnAttr, ArrayRef<int64_t> tileSizes) {
+    llvm::StringRef numWorkgroupsFnAttr, ArrayRef<int64_t> tileSizes,
+    ArrayRef<int64_t> loopIndices) {
+  assert(loopIndices.size() == 3);
```

Comment:
I dont think we always have loopIndices.size() == 3. Maybe loopIndices.size() <= 3 ?

Also might need some more checks
1) The corresponding tileSizes are set to 0
2) That the max value in `loopIndices` is less than `parallelLoopRange.size()`. Might be better to prune `loopIndices` accordingly

Reply:
`loopIndices` is used for selecting the three dimensions for distribution. We will always have three dimensions for distribution. (The default is just `{0, 1, 2}` if the launch configuration did not specify it explicitly.) If an op is not really distributed along all three dimensions then the trailing numbers will just be ignored. Added more docs.